segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java

```diff
@@ -0,0 +1,253 @@
+     *
+     * @return The maximum length of any key, in bytes.
+     */
+    int maximumKeyLength();
+
+    /**
+     * Gets a value indicating the maximum length of any Table Entry Value supported by this TableStore Implementation.
+     *
+     * @return The maximum length of any value, in bytes.
+     */
+    int maximumValueLength();
+
+    /**
+     * Creates a new Segment and marks it as a Table Segment.
+     * This segment may not be used for Streaming purposes (i.e., it cannot be used with {@link StreamSegmentStore}).
+     *
+     * @param segmentName The name of the Table Segment to create.
+     * @param timeout     Timeout for the operation.
+     * @return A CompletableFuture that, when completed normally, will indicate the operation completed. If the operation
+     * failed, the future will be failed with the causing exception. Notable Exceptions:
+     * <ul>
+     * <li>{@link StreamSegmentExistsException} If the Segment does exist (whether as a Table Segment or Stream Segment).
+     * </ul>
+     */
+    CompletableFuture<Void> createSegment(String segmentName, Duration timeout);
+
+    /**
+     * Deletes an existing Table Segment.
+     *
+     * @param segmentName The name of the Table Segment to delete.
+     * @param mustBeEmpty If true, the Table Segment will only be deleted if it is empty (contains no keys).
+     * @param timeout     Timeout for the operation.
+     * @return A CompletableFuture that, when completed normally, will indicate the operation completed. If the operation
+     * failed, the future will be failed with the causing exception. Notable Exceptions:
+     * <ul>
+     * <li>{@link StreamSegmentNotExistsException} If the Table Segment does not exist.
+     * <li>{@link BadSegmentTypeException} If segmentName refers to a non-Table Segment.
+     * <li>{@link TableSegmentNotEmptyException} If mustBeEmpty is true and the Table Segment contains at least one key.
+     * </ul>
+     */
+    CompletableFuture<Void> deleteSegment(String segmentName, boolean mustBeEmpty, Duration timeout);
+
+    /**
+     * Merges a Table Segment into another Table Segment.
+     *
+     * @param targetSegmentName The name of the Table Segment to merge into.
+     * @param sourceSegmentName The name of the Table Segment to merge.
+     * @param timeout           Timeout for the operation.
+     * @return A CompletableFuture that, when completed normally, will indicate the operation completed. If the operation
+     * failed, the future will be failed with the causing exception. Notable Exceptions:
+     * <ul>
+     * <li>{@link StreamSegmentNotExistsException} If either the Source or Target Table Segment do not exist.
+     * <li>{@link BadSegmentTypeException} If sourceSegmentName or targetSegmentName refer to non-Table Segments.
+     * </ul>
+     */
+    CompletableFuture<Void> merge(String targetSegmentName, String sourceSegmentName, Duration timeout);
```

Comment:
Say we have the same key _K_ present in both source and target. During the processing of the merge, what value of _K_ do we keep, is it the one of the target? I suppose that this is what merge aims to do, performing a bulk update using the key-value pairs in the target, but please clarify.

Reply:
When implemented in the future, this will work exactly as regular segment mergers. The txn will be appended as one atomic block at the end of the target segment and the values from the txn will prevail.